### PR TITLE
Added symlink support to path traversal detection

### PR DESCRIPTION
**Pull request by abhaynayar**
### Describe the Pull Request

```
In Ubuntu 18.04:

If MobSF is running in a directory accessed by a symlink, under
static analysis of an app, opening a file leads to "Path Traversal
Detected!" error.

Under MobSF/utils.py: is_safe_path() the path traversal check
fails due to absence of os.path.realpath() applied on the safe_root.

Therefore if ~/shared is a symlink to /media/shared then one
evaluates to /home/user/shared and other to /media/shared even
though they are the same directory.

Fixed by adding os.path.realpath() to both paths. Works on Linux.
```

### Checklist for PR

- [ ] Run MobSF unit tests and lint `tox -e lint,test`.
- [ ] Tested Working on Linux, Mac, Windows, and Docker
- [x] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)

### Additional Comments (if any)

```
This is my first open-source contribution, sorry for any glaring mistakes.
```

### CODE REVIEW

**Comment by ajinabraham** [comment]
I need to test this out, will revert soon.

**Comment by ajinabraham** [comment]
@abhaynayar Can you fix the lint errors?

File: MobSF/utils.py [hunk 1]
```diff
@@ -631,7 +631,7 @@ def is_safe_path(safe_root, check_path):
     safe_root = os.path.normpath(safe_root)
     check_path = os.path.normpath(check_path)
     return os.path.commonprefix(
-        (os.path.realpath(check_path), safe_root)) == safe_root
+        (os.path.realpath(check_path), os.path.realpath(safe_root))) == os.path.realpath(safe_root)
```
Comment (ajinabraham) [inline MobSF/utils.py]: I will suggest that you do this on the line before when variable is initialized.
Something like
```
safe_root = os.path.realpath(os.path.normpath(safe_root))
check_path = os.path.realpath(os.path.normpath(check_path))
```

**Comment by abhaynayar** [comment]
I'll look into the lint errors.

File: MobSF/utils.py [hunk 2]
```diff
@@ -628,10 +628,10 @@ def get_proxy_ip(identifier):
 
 def is_safe_path(safe_root, check_path):
     """Detect Path Traversal."""
-    safe_root = os.path.normpath(safe_root)
-    check_path = os.path.normpath(check_path)
+    safe_root = os.path.realpath(os.path.normpath(safe_root))
+    check_path = os.path.realpath(os.path.normpath(check_path))
     return os.path.commonprefix(
-        (os.path.realpath(check_path), safe_root)) == safe_root
+        (os.path.realpath(check_path), safe_root))) == safe_root)
```
Comment (ajinabraham) [inline MobSF/utils.py]: Why `realpath()` again?
See: https://mobsf.github.io/docs/#/tests for linting and running tests locally

File: MobSF/utils.py [hunk 3]
```diff
@@ -628,11 +628,9 @@ def get_proxy_ip(identifier):
...
 def is_safe_path(safe_root, check_path):
     """Detect Path Traversal."""
-    safe_root = os.path.normpath(safe_root)
-    check_path = os.path.normpath(check_path)
-    return os.path.commonprefix(
-        (os.path.realpath(check_path), safe_root)) == safe_root
-
+    safe_root = os.path.realpath(os.path.normpath(safe_root))
+    check_path = os.path.realpath(os.path.normpath(check_path))
+    return os.path.commonprefix((check_path, safe_root)) == safe_root
```
Comment (ajinabraham) [inline MobSF/utils.py]: Please remove extra brackets. it's not required.

Comment (abhaynayar) [reply MobSF/utils.py]: If you are talking about this:
```suggestion
    return os.path.commonprefix(check_path, safe_root) == safe_root
```

Then it is actually a tuple, so it is required. I first removed and checked, it wasn't working.

Comment (ajinabraham) [reply MobSF/utils.py]: It should be a list or path-like object. See: https://docs.python.org/3/library/os.path.html#os.path.commonprefix

**Comment by abhaynayar** [comment]
`tox -e lint` is working fine:


```
$ sudo tox -e lint
lint recreate: /████/Mobile-Security-Framework-MobSF/.tox/lint
lint installdeps: pydocstyle, flake8, flake8-broken-line, flake8-bugbear, flake8-builtins, flake8-colors, flake8-commas, flake8-comprehensions, flake8-docstrings, flake8-eradicate, flake8-import-order, flake8-logging-format, flake8-quotes, flake8-self, pep8-naming, radon
lint installed: attrs==19.3.0,colorama==0.4.1,entrypoints==0.3,eradicate==1.0,flake8==3.7.9,flake8-broken-line==0.2.0,flake8-bugbear==20.1.4,flake8-builtins==1.5.2,flake8-colors==0.1.6,flake8-commas==2.0.0,flake8-comprehensions==3.2.2,flake8-docstrings==1.5.0,flake8-eradicate==0.3.0,flake8-import-order==0.18.1,flake8-logging-format==0.6.0,flake8-polyfill==1.0.2,flake8-quotes==3.0.0,flake8-self==0.2.2,future==0.18.2,importlib-metadata==1.5.2,mando==0.6.4,mccabe==0.6.1,pep8-naming==0.10.0,pycodestyle==2.5.0,pydocstyle==5.0.2,pyflakes==2.1.1,radon==4.1.0,six==1.14.0,snowballstemmer==2.0.0,zipp==3.1.0
lint run-test-pre: PYTHONHASHSEED='3986433031'
lint run-test: commands[0] | python3.7 /████/Mobile-Security-Framework-MobSF/.tox/lint/bin/flake8
__________________________________________________________________ summary ___________________________________________________________________
  lint: commands succeeded
  congratulations :)

```


 But `tox -e test` is throwing up some errors, do I need to check that as well?

File: MobSF/utils.py (hunk 3 above)
Comment (abhaynayar) [reply MobSF/utils.py]: Okay so should I change the tuple to a list?
```suggestion
    return os.path.commonprefix([check_path, safe_root]) == safe_root
```

Comment (ajinabraham) [reply MobSF/utils.py]: Right

**Comment by ajinabraham** [comment]
Thanks for the PR. merging to master!

**Pull request merged**